packages/flagging/src/openfeature/provider.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { StandardResolutionReasons, type EvaluationContext, type Logger } from '@openfeature/core'
+import { createDatadogProvider } from './provider'
+
+describe('DatadogProvider', () => {
+  let provider: ReturnType<typeof createDatadogProvider>
+  let mockLogger: Logger
+  let mockContext: EvaluationContext
+
+  beforeEach(() => {
+    provider = createDatadogProvider()
+    mockLogger = {
+      debug: jasmine.createSpy('debug'),
+      info: jasmine.createSpy('info'),
+      warn: jasmine.createSpy('warn'),
+      error: jasmine.createSpy('error'),
+    }
+    mockContext = {}
+  })
+
+  describe('metadata', () => {
+    it('should have correct metadata', () => {
+      expect(provider.metadata).toEqual({
+        name: 'datadog',
+      })
+    })
+
+    it('should run on client', () => {
+      expect(provider.runsOn).toBe('client')
+    })
+  })
+
+  describe('resolveBooleanEvaluation', () => {
+    it('should return default value with DEFAULT reason', () => {
+      const result = provider.resolveBooleanEvaluation('test-flag', true, mockContext, mockLogger)
```

Comment:
Interesting they want `logger` as a fourth argument ([package docs](https://www.npmjs.com/package/@openfeature/web-sdk#extending)). It seems it (and `context`) can be set at the provider level. Is this just for extra flexibility?

Reply:
@aarsilv I don't see a way to set logger on provider level (in provider API). For context, I guess that helps with context switches and it helps for providers in [dynamic-context paradigm](https://openfeature.dev/specification/glossary#dynamic-context-paradigm).

Reply:
What's even funnier about logger is that OF [discourages](https://openfeature.dev/specification/sections/flag-evaluation#requirement-1411) any logging in resolve methods